nacos V3.1.1 Starting issue

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'configChangeClusterSyncRequestHandler' defined in URL [jar:nested:/G:/test/nacos-server-3.1.1/nacos/target/nacos-server.jar/!BOOT-INF/lib/nacos-config-3.1.1.jar!/com/alibaba/nacos/config/server/remote/ConfigChangeClusterSyncRequestHandler.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'configMigrateService': Invocation of init method failed
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:804)
at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:240)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1395)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1222)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1188)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1123)
at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
at com.alibaba.nacos.bootstrap.NacosBootstrap.startCoreContext(NacosBootstrap.java:96)
at com.alibaba.nacos.bootstrap.NacosBootstrap.startWithConsole(NacosBootstrap.java:84)
at com.alibaba.nacos.bootstrap.NacosBootstrap.main(NacosBootstrap.java:54)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:102)
at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64)
at org.springframework.boot.loader.launch.PropertiesLauncher.main(PropertiesLauncher.java:580)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'configMigrateService': Invocation of init method failed
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:222)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:429)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1818)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:607)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:315)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1752)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1635)
at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:913)
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
... 26 common frames omitted
Caused by: java.lang.Exception: [migrate] config_info namespace migrate pre check failed
at com.alibaba.nacos.config.server.service.ConfigMigrateService.namespaceMigratePreCheck(ConfigMigrateService.java:765)
at com.alibaba.nacos.config.server.service.ConfigMigrateService.doCheckNamespaceMigrate(ConfigMigrateService.java:519)
at com.alibaba.nacos.config.server.service.ConfigMigrateService.migrate(ConfigMigrateService.java:158)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMethod.invoke(InitDestroyAnnotationBeanPostProcessor.java:457)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:401)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:219)
... 41 common frames omitted
2026-03-17 17:29:34,108 ERROR Nacos failed to start, please see G:\test\nacos-server-3.1.1\nacos\logs\nacos.log for more details.

2026-03-17 17:29:34,118 INFO

Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.

2026-03-17 17:29:34,135 ERROR Application run failed

org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'configChangeClusterSyncRequestHandler' defined in URL [jar:nested:/G:/test/nacos-server-3.1.1/nacos/target/nacos-server.jar/!BOOT-INF/lib/nacos-config-3.1.1.jar!/com/alibaba/nacos/config/server/remote/ConfigChangeClusterSyncRequestHandler.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'configMigrateService': Invocation of init method failed
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:804)
at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:240)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1395)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1232)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:569)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.instantiateSingleton(DefaultListableBeanFactory.java:1222)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingleton(DefaultListableBeanFactory.java:1188)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:1123)
at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:987)
at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:627)
at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:752)
at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:439)
at org.springframework.boot.SpringApplication.run(SpringApplication.java:318)
at org.springframework.boot.builder.SpringApplicationBuilder.run(SpringApplicationBuilder.java:149)
at com.alibaba.nacos.bootstrap.NacosBootstrap.startCoreContext(NacosBootstrap.java:96)
at com.alibaba.nacos.bootstrap.NacosBootstrap.startWithConsole(NacosBootstrap.java:84)
at com.alibaba.nacos.bootstrap.NacosBootstrap.main(NacosBootstrap.java:54)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:102)
at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:64)
at org.springframework.boot.loader.launch.PropertiesLauncher.main(PropertiesLauncher.java:580)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'configMigrateService': Invocation of init method failed
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:222)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.applyBeanPostProcessorsBeforeInitialization(AbstractAutowireCapableBeanFactory.java:429)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1818)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:607)
at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:529)
at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:339)
at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:373)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:337)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:315)
at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1752)
at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1635)
at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:913)
at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791)
... 26 common frames omitted
Caused by: java.lang.Exception: [migrate] config_info namespace migrate pre check failed
at com.alibaba.nacos.config.server.service.ConfigMigrateService.namespaceMigratePreCheck(ConfigMigrateService.java:765)
at com.alibaba.nacos.config.server.service.ConfigMigrateService.doCheckNamespaceMigrate(ConfigMigrateService.java:519)
at com.alibaba.nacos.config.server.service.ConfigMigrateService.migrate(ConfigMigrateService.java:158)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMethod.invoke(InitDestroyAnnotationBeanPostProcessor.java:457)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:401)
at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:219)
... 41 common frames omitted
2026-03-17 17:29:34,137 INFO [ThreadPoolManager] Start destroying ThreadPool

2026-03-17 17:29:34,137 INFO [ThreadPoolManager] Completed destruction of ThreadPool

When using the nacos-plugin xugu data plugin to start V3.1.1, an error occurred! However, if the configuration is changed to use the MySQL data source, there will be no problem.

Starting the Windows 10 console

The Nacos-plugin project is using the code from the develop branch. Currently, I am working on adapting xuguDB to 3.1.x.

--- a/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/AbstractMapperByXugu.java
+++ b/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/AbstractMapperByXugu.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright 1999-2022 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.impl.xugu;
+
+import com.alibaba.nacos.plugin.datasource.emums.TrustedXuguFunctionEnum;
+import com.alibaba.nacos.plugin.datasource.mapper.AbstractMapper;
+
+/**
+ * The abstract Xugu mapper contains CRUD methods.
+ *
+ * @author Ken
+ */
+public abstract class AbstractMapperByXugu extends AbstractMapper {
+
+    @Override
+    public String getFunction(String functionName) {
+        return TrustedXuguFunctionEnum.getFunctionByName(functionName);
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/ConfigInfoGrayMapperByXugu.java
+++ b/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/ConfigInfoGrayMapperByXugu.java
@@ -1,0 +1,44 @@
+/*
+ * Copyright 1999-2022 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.impl.xugu;
+
+import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.mapper.ConfigInfoGrayMapper;
+import com.alibaba.nacos.plugin.datasource.model.MapperContext;
+import com.alibaba.nacos.plugin.datasource.model.MapperResult;
+
+import java.util.Collections;
+
+/**
+ * The XuguDB implementation of ConfigInfoGrayMapper.
+ *
+ * @author Ken
+ */
+public class ConfigInfoGrayMapperByXugu extends AbstractMapperByXugu implements ConfigInfoGrayMapper {
+
+    @Override
+    public MapperResult findAllConfigInfoGrayForDumpAllFetchRows(MapperContext context) {
+        String sql = "SELECT id, data_id, group_id, tenant_id, gray_name, gray_rule, app_name, content, md5, gmt_modified "
+                + "FROM config_info_gray ORDER BY id LIMIT " + context.getPageSize() + " OFFSET " + context.getStartRow();
+        return new MapperResult(sql, Collections.emptyList());
+    }
+
+    @Override
+    public String getDataSource() {
+        return DatabaseTypeConstant.XUGU;
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/ConfigMigrateMapperByXugu.java
+++ b/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/java/com/alibaba/nacos/plugin/datasource/impl/xugu/ConfigMigrateMapperByXugu.java
@@ -1,0 +1,33 @@
+/*
+ * Copyright 1999-2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.nacos.plugin.datasource.impl.xugu;
+
+import com.alibaba.nacos.plugin.datasource.constants.DatabaseTypeConstant;
+import com.alibaba.nacos.plugin.datasource.mapper.ConfigMigrateMapper;
+
+/**
+ * The XuguDB implementation of ConfigMigrateMapper.
+ *
+ * @author Ken
+ */
+public class ConfigMigrateMapperByXugu extends AbstractMapperByXugu implements ConfigMigrateMapper {
+
+    @Override
+    public String getDataSource() {
+        return DatabaseTypeConstant.XUGU;
+    }
+}

--- a/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/resources/META-INF/services/com.alibaba.nacos.plugin.datasource.mapper.Mapper
+++ b/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/resources/META-INF/services/com.alibaba.nacos.plugin.datasource.mapper.Mapper
@@ -15,8 +15,10 @@
 #
 
 com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigInfoBetaMapperByXugu
+com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigInfoGrayMapperByXugu
 com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigInfoMapperByXugu
 com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigInfoTagMapperByXugu
+com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigMigrateMapperByXugu
 com.alibaba.nacos.plugin.datasource.impl.xugu.ConfigTagsRelationMapperByXugu
 com.alibaba.nacos.plugin.datasource.impl.xugu.GroupCapacityMapperByXugu
 com.alibaba.nacos.plugin.datasource.impl.xugu.HistoryConfigInfoMapperByXugu

--- a/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/resources/schema/nacos-xugu.sql
+++ b/nacos-datasource-plugin-ext/nacos-xuguDB-datasource-plugin-ext/src/main/resources/schema/nacos-xugu.sql
@@ -40,6 +40,32 @@ CREATE TABLE `config_info`
     CONSTRAINT `uk_configinfo_datagrouptenant` UNIQUE (`data_id`, `group_id`, `tenant_id`)
 )COMMENT 'config_info';
 
+
+/******************************************/
+/*   表名称 = config_info_gray             */
+/******************************************/
+CREATE TABLE `config_info_gray` (
+    `id` bigint AUTO_INCREMENT NOT NULL COMMENT 'id',
+    `data_id` varchar(255) NOT NULL COMMENT 'data_id',
+    `group_id` varchar(128) NOT NULL COMMENT 'group_id',
+    `content` varchar NOT NULL COMMENT 'content',
+    `md5` varchar(32) DEFAULT NULL COMMENT 'md5',
+    `src_user` text COMMENT 'src_user',
+    `src_ip` varchar(100) DEFAULT NULL COMMENT 'src_ip',
+    `gmt_create` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'gmt_create',
+    `gmt_modified` datetime NOT NULL DEFAULT CURRENT_TIMESTAMP COMMENT 'gmt_modified',
+    `app_name` varchar(128) DEFAULT NULL COMMENT 'app_name',
+    `tenant_id` varchar(128) DEFAULT '' COMMENT 'tenant_id',
+    `gray_name` varchar(128) NOT NULL COMMENT 'gray_name',
+    `gray_rule` varchar NOT NULL COMMENT 'gray_rule',
+    `encrypted_data_key` varchar(256) NOT NULL DEFAULT '' COMMENT 'encrypted_data_key',
+    PRIMARY KEY (`id`),
+    CONSTRAINT `uk_configinfogray_datagrouptenantgray` UNIQUE (`data_id`,`group_id`,`tenant_id`,`gray_name`)
+)COMMENT 'config_info_gray';
+CREATE INDEX `idx_dataid_gmt_modified` on `config_info_gray` (`data_id`,`gmt_modified`);
+CREATE INDEX `idx_gmt_modified` on `config_info_gray` (`gmt_modified`);
+
+
 /******************************************/
 /*   表名称 = config_info_aggr             */
 /******************************************/

